RedHatInsights/rhsm-subscriptions
# ENT-4578: Handle lookup for On Demand subscriptions without capacities

Test Steps:

- Insert data:
`INSERT INTO public.offering(
    sku, product_name, product_family, sla, usage, description, has_unlimited_usage)
    VALUES ('MW01882', 'OpenShift Streams for Apache Kafka', 'JBoss', 'Premium', 'Production', 'Red Hat OpenShift Streams for Apache Kafka (Hourly)', false);

INSERT INTO public.offering(
    sku, product_name, product_family, sla, usage, description, has_unlimited_usage)
    VALUES ('MW01882RN', 'OpenShift Streams for Apache Kafka', 'JBoss', 'Standard', 'Disaster Recovery', Red Hat OpenShift Streams for Apache Kafka (Hourly)', false);


INSERT INTO public.subscription(
    sku, owner_id, subscription_id, quantity, start_date, end_date, billing_provider_id, account_number, subscription_number)
    VALUES ('MW01882', 'org123', '6872249', '1', '2021-11-30T05:00:00Z', '2022-11-30T04:59:59Z', '61a65565dec7af0ecb60119a', '1460290', 6872263);


INSERT INTO public.subscription(
    sku, owner_id, subscription_id, quantity, start_date, end_date, billing_provider_id, account_number, subscription_number)
    VALUES ('MW01882RN', 'org123', '3515143', '3', '2021-11-30T05:00:00Z', '2022-11-30T04:59:59Z', '61a65565dec7af0ecb60119a', '1460290', 123511);
`
- Start app with: `DEV_MODE=true ./gradlew :bootRun`
- Run curl command: `curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/subscriptions/products/rhosak' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='`
 - Should get back both subscriptions
